service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/SpecimenMapper.java

```diff
@@ -188,6 +189,16 @@ private boolean dummySpecimenOrObservationExists(Specimen specimen, List<Observa
             if (collection.hasBodySite()) {
                 specimenNarrativeStatementCommentBuilder.collectionSite(collection.getBodySite());
             }
+
+            if (collection.hasCollector()) {
+                var collector = messageContext.getInputBundleHolder().getResource(collection.getCollector().getReferenceElement());
+                collector.ifPresent(collectorPractitioner -> {
+                    if (collectorPractitioner.getResourceType().equals(ResourceType.Practitioner)) {
```

Comment:
I will leave it up to the author, but this could be written as a single monad chain:
```java
if (collection.hasCollector()) {
	messageContext.getInputBundleHolder().getResource(collection.getCollector().getReferenceElement())
		.filter(collector -> collector.getResourceType() == ResourceType.Practitioner)
		.map(collector -> buildHumanName((Practitioner) collector))
		.ifPresent(specimenNarrativeStatementCommentBuilder::collector);
}
```